packages/react-scripts/fixtures/kitchensink/src/App.js

```diff
@@ -1,23 +1,22 @@
-import React from 'react';
+import React, { Component, Children, cloneElement } from 'react';
 
 class BuiltEmitter extends React.Component {
-  constructor(props) {
-    super(props)
-
-    this.callWhenDone = done => done();
+  componentDidMount() {
+    const only = Children.only(this.props.children);
+    if (!Component.isPrototypeOf(only)) {
```

Comment:
Ah, good catch -- you're correct. I took this snippet from my proposal before @EnoahNetzach's superseded it and I was originally checking it before mounting the component (when both become elements).